flipt-engine/proto/evaluation.proto

```diff
@@ -0,0 +1,200 @@
+  ErrorEvaluationReason reason = 3;
+}
+
+service EvaluationService {
+  rpc Boolean(EvaluationRequest) returns (BooleanEvaluationResponse) {}
+  rpc Variant(EvaluationRequest) returns (VariantEvaluationResponse) {}
+  rpc Batch(BatchEvaluationRequest) returns (BatchEvaluationResponse) {}
+}
+
+message EvaluationDistribution {
+  string id = 1;
+  string rule_id = 2;
+  string variant_id = 3;
+  string variant_key = 4;
+  string variant_attachment = 5;
+  float rollout = 6;
+}
+
+enum EvaluationRolloutType {
+  UNKNOWN_ROLLOUT_TYPE = 0;
+  SEGMENT_ROLLOUT_TYPE = 1;
+  THRESHOLD_ROLLOUT_TYPE = 2;
+}
+
+message EvaluationRollout {
+  EvaluationRolloutType type = 1;
+  int32 rank = 2;
+  oneof rule {
+    EvaluationRolloutSegment segment = 3;
+    EvaluationRolloutThreshold threshold = 4;
+  }
+}
+
+message EvaluationRolloutThreshold {
+  float percentage = 1;
+  bool value = 2;
+}
+
+enum EvaluationSegmentOperator {
+  OR_SEGMENT_OPERATOR = 0;
+  AND_SEGMENT_OPERATOR = 1;
+}
+
+message EvaluationRolloutSegment {
+  bool value = 1;
+  EvaluationSegmentOperator segment_operator = 2;
+  repeated EvaluationSegment segments = 3;
+}
+
+enum EvaluationSegmentMatchType {
+  ALL_SEGMENT_MATCH_TYPE = 0;
+  ANY_SEGMENT_MATCH_TYPE = 1;
+}
+
+message EvaluationSegment {
+  string key = 1;
+  string name = 2;
+  string description = 3;
+  EvaluationSegmentMatchType match_type = 4;
+  google.protobuf.Timestamp created_at = 5;
+  google.protobuf.Timestamp updated_at = 6;
+  repeated EvaluationConstraint constraints = 7;
+}
+
+enum EvaluationFlagType {
+  VARIANT_FLAG_TYPE = 0;
+  BOOLEAN_FLAG_TYPE = 1;
+}
+
+message EvaluationFlag {
+  string key = 1;
+  string name = 2;
+  string description = 3;
+  bool enabled = 4;
+  EvaluationFlagType type = 5;
+  google.protobuf.Timestamp created_at = 6;
+  google.protobuf.Timestamp updated_at = 7;
+  repeated EvaluationRule rules = 8;
+  repeated EvaluationRollout rollouts = 9;
+}
+
+enum EvaluationConstraintComparisonType {
+  UNKNOWN_CONSTRAINT_COMPARISON_TYPE = 0;
+  STRING_CONSTRAINT_COMPARISON_TYPE = 1;
+  NUMBER_CONSTRAINT_COMPARISON_TYPE = 2;
+  BOOLEAN_CONSTRAINT_COMPARISON_TYPE = 3;
+  DATETIME_CONSTRAINT_COMPARISON_TYPE = 4;
+}
+
+message EvaluationConstraint {
+  string id = 1;
+  EvaluationConstraintComparisonType type = 2;
+  string property = 3;
+  string operator = 4;
+  string value = 5;
+}
+
+message EvaluationRule {
+  string id = 1;
+  repeated EvaluationSegment segments = 2;
+  int32 rank = 3;
+  EvaluationSegmentOperator segment_operator = 4;
+  repeated EvaluationDistribution distributions = 5;
+}
+
+message EvaluationNamespace {
+  string key = 1;
+}
+
+message EvaluationNamespaceSnapshot {
+  EvaluationNamespace namespace = 1;
+  repeated EvaluationFlag flags = 2;
+}
+
+message EvaluationNamespaceSnapshotRequest {
+  string key = 1;
+}
+
+// flipt:sdk:ignore
+service DataService {
+  rpc EvaluationSnapshotNamespace(EvaluationNamespaceSnapshotRequest) returns (EvaluationNamespaceSnapshot);
+}
```

Comment:
This file could potentially be a moving target if things change in the original repository, just to point it out.

Reply:
Yep, ideally it would be to have this in `flipt-rust` client and use it in this project. What do you think about this @markphelps?

Reply:
yeah, we're actively refactoring our Server side SDKs at the moment (rust included), so I would hold off for now

Also the `DataService` is really meant to be 'internal' in that users shouldn't depend on its structure (although we will always aim for backward compatibility in reality because protos).

tl;dr I think this is fine for now